speaker/layer2_controller.go

```diff
@@ -122,8 +123,10 @@ func (c *layer2Controller) ShouldAnnounce(l log.Logger, name string, toAnnounce
 	return "notOwner"
 }
 
-func (c *layer2Controller) SetBalancer(l log.Logger, name string, lbIP net.IP, pool *config.Pool) error {
-	c.announcer.SetBalancer(name, lbIP)
+func (c *layer2Controller) SetBalancer(l log.Logger, name string, lbIPs []net.IP, pool *config.Pool) error {
+	for _, lbIP := range lbIPs {
+		c.announcer.SetBalancer(name, lbIP)
```

Comment:
I missed this before, but this won't work. The L2 controller assumes that a given service holds only one IP.
See when it does
```
	a.ips[name] = ip
```
I think it's fine deferring dual stack support for L2 outside of this PR, it's already hard to review.

Reply:
i fixed this and make it a slice and added unit test case with svc and multiple ips